RELEASES.md

```diff
@@ -1,3 +1,116 @@
+Version 1.78.0 (2024-05-02)
+==========================
+
+<a id="1.78.0-Language"></a>
+
+Language
+--------
+- [Stabilize `#[cfg(target_abi = ...)]`](https://github.com/rust-lang/rust/pull/119590/)
+- [Stabilize the `#[diagnostic]` namespace and `#[diagnostic::on_unimplemented]` attribute](https://github.com/rust-lang/rust/pull/119888/)
+- [Make async-fn-in-trait implementable with concrete signatures](https://github.com/rust-lang/rust/pull/120103/)
+- [Make matching on NaN a hard error, and remove the rest of illegal_floating_point_literal_pattern](https://github.com/rust-lang/rust/pull/116284/)
+- [static mut: allow mutable reference to arbitrary types, not just slices and arrays](https://github.com/rust-lang/rust/pull/117614/)
+- [Add `non_local_definitions` lint for non-local `impl` definitions in functions](https://github.com/rust-lang/rust/pull/120393/)
```

Comment:
```suggestion
```

This was partially reverted[^1] in #122107.

[^1]: It was made allow-by-default (instead of warn-by-default)